### PR TITLE
feat(dpp): implement hashed ECDSA key type

### PR DESCRIPTION
**Pull request by shuplenkov**
## Issue being fixed or feature implemented

In order to support masternode owner key which hash160 of ECDSA we need to introduce a new identity public key type 2 and support it for singing and verification of State Transitions

## What was done?

Added new key type 2.

## How Has This Been Tested?

With tests

## Breaking Changes

No

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/js-dpp/test/unit/identity/IdentityPublicKey.spec.js [hunk 1]
```diff
@@ -118,6 +118,24 @@ describe('IdentityPublicKey', () => {
       expect(result).to.deep.equal(expectedHash);
     });
 
+    it('should return data in case ECDSA_HASH160', () => {
```
Comment (shumkov) [inline packages/js-dpp/test/unit/identity/IdentityPublicKey.spec.js:121]: Not sure I understand the point of this test

Comment (shuplenkov) [reply packages/js-dpp/test/unit/identity/IdentityPublicKey.spec.js:121]: in case of ECDSA_HASH160 we return data instead of data hash

Comment (shumkov) [reply packages/js-dpp/test/unit/identity/IdentityPublicKey.spec.js:121]: Oh... make sense

**Review by shumkov — APPROVED** [review]
LGTM 👍

File: packages/js-dpp/schema/identity/publicKey.json [hunk 2]
```diff
@@ -12,9 +12,10 @@
...
         0,
-        1
+        1,
+        2
       ],
-      "description": "Public key type. 0 - ECDSA Secp256k1, 1 - BLS 12-381",
+      "description": "Public key type. 0 - ECDSA Secp256k1, 1 - BLS 12-381, 2 - ECDSA Secp256k1 Hash160",
       "$comment": "It can't be changed after adding a key"
     },
     "purpose": {
```
Comment (shumkov) [inline packages/js-dpp/schema/identity/publicKey.json:21]: We need to define a schema for `data` property for type 2 (see `allOf`)

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/js-dpp/schema/identity/publicKey.json [hunk 3]
```diff
@@ -82,6 +83,27 @@
...
+        }
+      },
+      "then": {
+        "properties": {
+          "data": {
+            "type": "array",
+            "byteArray": true,
+            "minItems": 20,
+            "maxItems": 20,
+            "description": "BLS public key Hash160",
```
Comment (shumkov) [inline packages/js-dpp/schema/identity/publicKey.json:102]: It's ECDSA Secp256k1

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/js-dpp/schema/identity/publicKey.json (hunk 3 above)
Comment (shuplenkov) [reply packages/js-dpp/schema/identity/publicKey.json]: Copy paste ;)

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**